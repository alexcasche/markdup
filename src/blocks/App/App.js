import styled from 'styled-components'

const App = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow-x: hidden;
`

export default App
