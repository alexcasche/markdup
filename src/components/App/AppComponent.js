import React, { Component } from 'react'
import { App } from 'blocks'
import { Dialog, Page, Settings, Sidebar } from 'components'
import { PropTypes as t } from 'prop-types'

export default class AppComponent extends Component {
  componentDidMount() {
    this.props.actions.appSetup()
  }
  componentDidUpdate(nextProps, nextState) {
    const { allSettings, actions } = nextProps
    if (allSettings.autoSave) {
      actions.saveSession()
    }
  }
  render() {
    return (
      <App className="App">
        <Sidebar />
        <Page />
        <Settings />
        <Dialog />
      </App>
    )
  }
}

AppComponent.propTypes = {
  allWorkspaces: t.object.isRequired,
  allSettings: t.object.isRequired,
  actions: t.shape({
    appSetup: t.func.isRequired,
    saveSession: t.func.isRequired,
  }).isRequired,
}
