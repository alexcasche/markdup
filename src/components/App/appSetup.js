import { getSession } from 'utils/localStorage'
import { initSettings, initWorkspaces } from 'actions'

export const appSetup = () => {
  const session = getSession()
  if (session) {
    return dispatch => {
      const { settings, workspaces } = session
      dispatch(initSettings(settings))
      dispatch(initWorkspaces(workspaces))
    }
  } else {
    return null
  }
}
