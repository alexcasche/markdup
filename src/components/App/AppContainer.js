import React from 'react'
import { connect } from 'react-redux'
import AppComponent from './AppComponent'
import { getAllSettings, getAllWorkspaces } from 'selectors'
import { appSetup } from './appSetup'
import { saveSession } from 'utils/localStorage'

export const AppContainer = props => {
  return <AppComponent {...props} />
}

function mapStateToProps(state) {
  return {
    allSettings: getAllSettings(state),
    allWorkspaces: getAllWorkspaces(state),
  }
}
function mapDispatchToProps(dispatch) {
  return {
    actions: {
      appSetup: appSetup,
      saveSession: saveSession,
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AppContainer)
