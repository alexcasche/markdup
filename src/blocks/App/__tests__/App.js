import App from '../App'

describe('App Block', () => {
  it('renders correctly', () => {
    const tree = render(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
