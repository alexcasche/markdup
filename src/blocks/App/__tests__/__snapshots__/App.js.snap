// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Block renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  overflow-x: hidden;
}

<div
  className="c0"
/>
`;
